### PR TITLE
Fix building with template-haskell-2.16+

### PR DESCRIPTION
**Pull request by mimi1vx**
Fixes #27

### CODE REVIEW

File: src/Text/Heterocephalus.hs [hunk 1]
```diff
@@ -522,7 +522,15 @@ docToExp set scope (DocCond conds final) = do
     go (d, docs) = do
       let d' = derefToExp ((specialOrIdent, VarE 'or) : scope) d
       docs' <- docsToExp set scope docs
-      return $ TupE [d', docs']
+      return $ nonUnaryTupE [d', docs']
+      where
+        nonUnaryTupE :: [Exp] -> Exp
+        nonUnaryTupE [e] = e
```
Comment (cdepillabout) [inline src/Text/Heterocephalus.hs]: It seems like this line would never match, since you're explicitly passing `nonUnaryTupE` two elements.

Comment (mimi1vx) [reply src/Text/Heterocephalus.hs]: true, we use TupE only once with exact elements :D

**Comment by arowM** [comment]
Thanks.
Could you please fix CI test failures on stack LTS-9?
https://travis-ci.org/github/arowM/heterocephalus/builds/731874759

**Comment by mimi1vx** [comment]
@arowM may I suggest focus on more actual stackage lts?  9.x is more than 3 years unsupported.

from actual haskell projects most used now is ghc 8.6.5 and then 8.8.x + many projects started moving till 8.10.2 --> LTS-14, LTS-14 and Stackage nightly

**Comment by arowM** [comment]
@mimi1vx 
OK. So, could you fix lower boundaries for dependencies in cabal file?

**Comment by mimi1vx** [comment]
@arowM https://travis-ci.org/github/mimi1vx/heterocephalus/jobs/731896688 this is LTS-9 job on current master .. so LTS9 is broken longer time

**Comment by arowM** [comment]
@mimi1vx 
It seems current master does not fails.
https://travis-ci.org/github/arowM/heterocephalus/builds/473045380

**Comment by mimi1vx** [comment]
> @mimi1vx
> It seems current master does not fails.
> https://travis-ci.org/github/arowM/heterocephalus/builds/473045380

this was true 2 years ago, if you rerun/restart job now .. it fail. which is with Stackage weird, but still

my link is job current master , with restarted travis.ci today

**Comment by arowM** [comment]
@mimi1vx 
Can you separate the commits about CI tool into a separate and independent PR?
It's because changing the CI tool has a different purpose than this PR.

**Comment by arowM** [comment]
Thank you.
I'll fix a few things and then release a new version.

**Comment by arowM** [comment]
Released [1.0.5.4](https://hackage.haskell.org/package/heterocephalus)

**Pull request merged**